extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -162,17 +227,29 @@ export class AmpSelector extends AMP.BaseElement {
   }
 
   /**
+   * Clears all selected options.
+   * @private
+   */
+  clearAllSelections_() {
+    while (this.selectedOptions_.length > 0) {
+      // Clear selected options for single select.
+      const el = this.selectedOptions_.pop();
+      this.clearSelection_(el);
+    }
+  }
+
+  /**
    * Marks a given element as selected and clears the others if required.
    * @param {!Element} element.
    * @private
    */
   setSelection_(element) {
+    // Exit if `element` is already selected.
+    if (this.selectedOptions_.indexOf(element) >= 0) {
+      return;
+    }
```

Comment:
im ok with this change but we should not be calling this on already selected elements :)